_posts/2020-08-17-creating-your-ci-cd-pipeline-with-concourse.md

```diff
@@ -0,0 +1,219 @@
+Also, Concourse is nothing new and fancy, it has been there since the year 2014.
+
+Anyway, because you still be reading this, you probably think as I do - 
+one cannot know if it's good or not, without trying it. So, let's get forward.
+
+Basic concepts
+----
+
+Concourse pipelines are defined with [YAML](https://yaml.org/). Core concepts
+are [resources](https://concourse-ci.org/resources.html), [tasks](https://concourse-ci.org/tasks.html) 
+and [jobs](https://concourse-ci.org/jobs.html). For every pipeline, there are inputs and outputs which are conceptually
+resource states with direction. Concourse itself doesn't know any details about these. Resources are just some external
+sources with state and which you can interact by using get and put operations. Each resource has a resource type.
+If a resource is an abstraction for some project artifact, e.g. your codebase in git at a certain point in time,
+then resource type is an abstraction of git as a whole. However, from the implementation point of view resource type
+could also be seen as an abstraction of an actual implementation of how your pipeline interacts with some specific resource,
+e.g. your git repository.
+
+But, this is computer science. Sometimes it feels like, the more vague some concept is, the more
+popular it is (you probably have heard of service-oriented architecture and can strictly define what is a service
+and what is not). Anyway, in the context of this blog post, we can just think that we have *things* where we get 
+something (input) and *things* where we put something (output). These can be external *things*, or 
+internal *things* to pass some result between tasks.
+
+There are some Concourse core resources available which you can use without defining resource type explicitly
+and [there are also some third-party resources listed on the Concourse website](https://resource-types.concourse-ci.org/).
+In addition, there are also many others in the wild Internet.
+
+However, **keep security concerns in mind** whenever you use third-party resource types and especially
+with the ones which are not listed on the Concourse website. Resource types are in practice just Docker images, so
+if you are unlucky you can find yourself pulling an image that is changed to something nasty without you knowing it.
+Fortunately, it's quite an easy job to make your own resource type. So, if you feel like
+you don't trust enough for the publisher of some resource type, you can use it as a base for your own resource type.
+However, remember to contribute to the base project if you make good improvements or bug fixes. 
+
+Let's build and deploy some dummy docker images into the Docker Hub 
+------------------
+
+I don't go into details on how to start and run Concourse, but I have provided some instructions
+in [my example project in Github](https://github.com/solita-alperttiti/concourse-ci-example).
+Also, a fully runnable pipeline example can be found from that project.
```

Comment:
Maybe the wording should be reversed so also could be dropped e.g "A fully runnable pipeline is included in the project for example purposes"

Reply:
I kind of like to keep it that way, because it attaches it more closely to that there are instructions and _also_ pipeline example. Without that it feels too separate. Anyway, I think this is matter of opinion and writing style.